E2E: reduce default timeout value to 10000ms

worldomonation opened the pull request:
#### Proposed Changes

This PR reduces the default timeout value for the Playwright side of the E2E framework to 10000ms.

The default timeout of the Jest Side remains at the default value.

Context: p1669068425400639-slack-C1A1EKDGQ

#### Testing Instructions

1. pull this branch.
2. open a test file and insert a failing wait statement (eg. `waitForURL('ljwer')`.
3. execute the test with `yarn jest ...`

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

worldomonation committed:
Add TIMEOUT and set every page to use the new TIMEOUT

karenroldan reviewed (commented):
**suggestion:** can we increase the default timeout value to 15s (15000ms)? 

I see that there are mobile E2E tests that failed due to timeout: `Timeout 10000ms exceeded.` 

I re-triggered the build for the mobile E2E test and there was still [1 failure due to 10s timeout](https://teamcity.a8c.com/repository/download/calypso_calypso_WebApp_Calypso_E2E_Playwright_mobile/9055007:id/screenshots/setup__build__land-in-home-dashboard__0-0.webm) while waiting to navigate to the given URL.

The test that failed is `specs/onboarding/setup__build.ts` - Land in Home dashboard.

karenroldan commented:
> Would you be open to keeping the shortened timeout if I can fix it?

Of course! Thank you for looking into it.

worldomonation committed:
packages/calypso-e2e/src/lib/pages/plugins-page.ts
- apply a default timeout of 15 seconds due to the `networkidle` wait.

test/e2e/specs/onboarding/setup__build.ts
- apply a default wait of 30 seconds due to the stepper being slow.

worldomonation committed:
Update comments on both files.

worldomonation commented:
Added some manual exceptions/overrides at the two locations where it was failing on Mobile viewport.

One is well-known - the `networkidle` call when loading Calypso on a Simple site may take longer than expected to fire. In the case of `/home` endpoint, the event may not fire for up to ~25 seconds.

The other is a bad UX on top of taking long time to load, and I've reached out to (hopefully) the right team on Slack. p1669398917141209-slack-C02T4NVL4JJ

karenroldan reviewed (approved):
LGTM 🚀  thanks for working on it!
